server/routes/api/stats.py

```diff
@@ -414,65 +414,212 @@ def get_place_stat_date_within_place():
     return Response(json.dumps(response), 200, mimetype='application/json')
 
 
-def get_set_csv_rows(stat_set, sv_list, row_limit=None):
+def get_facets_by_variable_series(series_response):
     """
-    Gets the csv rows for a set of statistical variables data of a single date.
+    Gets the available facets for each sv in an api response for series_within.
 
     Args:
-        stat_set: set of statistical variable data points with the format: 
+        series_response: the response from a dc.series_within call.
+    
+    Returns:
+        a dict of sv to dict of facet id to facet information:
             {
-                data: {
-                    [svDcid]: {
-                        stat: {
-                            [placeDcid]: {
-                                date: "",
-                                value: "",
-                                metaHash: ""
-                            } ,
-                            ...
-                        }
+                [sv]: {
+                    [facet1]: {
+                        "importName": "Census",
+                        "observationPeriod": "P1Y",
+                        ...
                     },
                     ...
                 },
-                metadata: {
-                    [metaHash]: {
-                        provenanceUrl: "",
+                ...
+            }
+    """
+    facets_by_variable = {}
+    facets = series_response.get("facets", {})
+    for sv_obs in series_response.get("observationsByVariable", []):
+        sv = sv_obs.get("variable", "")
+        facets_by_variable[sv] = {}
+        for place_obs in sv_obs.get("observationsByEntity", []):
+            for facet_obs in place_obs.get("seriesByFacet", []):
+                facet = str(facet_obs.get("facet", ""))
+                if facet and not facet in facets_by_variable[sv]:
+                    facets_by_variable[sv][facet] = facets.get(facet, {})
+    return facets_by_variable
+
+
+def get_facets_by_variable_points(points_response):
```

Comment:
nit: rename to: get_variable_facets_from_points()

Reply:
done